### PR TITLE
posix: consolidate include paths

### PR DESCRIPTION
**Pull request by authmillenon**
Though I introduced this split myself, I see no specific reason for doing it this way and find it rather tidious. Note that we can't include the POSIX headers into sys/include, since otherwise POSIX applications would not build on native.

### CODE REVIEW

**Comment by authmillenon** [comment]
Rebased to current master

File: pkg/libcoap/0001-Add-RIOT-Makefile.patch [hunk 1]
```diff
@@ -15,7 +15,7 @@ index 0000000..f90baa1
 +++ b/Makefile
 @@ -0,0 +1,5 @@
 +MODULE:=$(shell basename $(CURDIR))
-+INCLUDES += -I$(RIOTBASE) -I$(RIOTBASE)/sys/include -I$(RIOTBASE)/core/include -I$(RIOTBASE)/drivers/include -I$(RIOTBASE)/drivers/cc110x_legacy/include -I$(RIOTBASE)/cpu/arm7_common/include -I$(RIOTBASE)/sys/net/transport_layer/include -I$(RIOTBASE)/sys/net/sixlowpan/include/ -I$(RIOTBASE)/sys/net/ieee802154/include -I$(RIOTBASE)/sys/net/net_help -I$(RIOTBASE)/sys/posix/include -I$(RIOTBASE)/sys/posix/pnet/include
++INCLUDES += -I$(RIOTBASE) -I$(RIOTBASE)/sys/include -I$(RIOTBASE)/core/include -I$(RIOTBASE)/drivers/include -I$(RIOTBASE)/drivers/cc110x_legacy/include -I$(RIOTBASE)/cpu/arm7_common/include -I$(RIOTBASE)/sys/net/transport_layer/include -I$(RIOTBASE)/sys/net/sixlowpan/include/ -I$(RIOTBASE)/sys/net/ieee802154/include -I$(RIOTBASE)/sys/net/net_help -I$(RIOTBASE)/sys/posix/include
```
Comment (Kijewski) [inline pkg/libcoap/0001-Add-RIOT-Makefile.patch]: This line should not be needed at all.

Comment (authmillenon) [inline pkg/libcoap/0001-Add-RIOT-Makefile.patch]: Addressed.

**Comment by Kijewski** [comment]
Two errors for native: https://travis-ci.org/RIOT-OS/RIOT/jobs/77329976#L1179

**Comment by authmillenon** [comment]
Then this PR is not mergable until `uart0` for native is dropped T.T. (Those apps include `shell`, for native this draws in `uart0` and thus `posix`, causing the include path for `dev_eth_tap.h` to look into the posix path, breaking the normal system includes there)

**Comment by authmillenon** [comment]
Alternatively we could move `socket_base` to `sys/include` for now (will be removed after #3615 got merged and the modules using `socket_base` are adapted anyways).

**Comment by authmillenon** [comment]
Adapted https://github.com/RIOT-OS/RIOT/pull/3611#issuecomment-136131249. Scream if you don't like that.

**Comment by authmillenon** [comment]
Rebased to current master and removed 07c2c1f as `uart0` was dropped from master.

**Comment by authmillenon** [comment]
Travis is happy now.

**Comment by cgundogan** [comment]
ACK. building for native and iotlab-m3 works without error. I added `USEPKG += libcoap` to test this to the `gnrc_networking` example

**Pull request merged**